crates/re_types/src/archetype.rs

```diff
@@ -24,15 +24,41 @@ use crate::Component;
 /// manually extend existing archetypes, or even implement fully custom ones.
 ///
 /// Most [`Archetype`] methods are optional to implement.
-/// The most important method to implement is [`Archetype::as_component_lists`], which describes how
-/// the archetype can be interpreted as a [`ComponentList`]: a set of components that are ready to
+/// The most important method to implement is [`Archetype::as_component_batches`], which describes how
+/// the archetype can be interpreted as a [`ComponentBatch`]: a set of components that are ready to
 /// be serialized.
 ///
 /// Have a look at our [Custom Data] example to learn more about handwritten archetypes.
 ///
 /// [IDL definitions]: https://github.com/rerun-io/rerun/tree/latest/crates/re_types/definitions/rerun
 /// [Custom Data]: https://github.com/rerun-io/rerun/blob/latest/examples/rust/custom_data/src/main.rs
 pub trait Archetype {
+    /// The associated indicator component, whose presence indicates that the high-level
+    /// archetype-based APIs were used to log the data.
+    ///
+    /// Indicator components open new opportunities in terms of API design, better heuristics and
+    /// performance improvements on the query side.
```

Comment:
(I know it's moved, but) that's more of a note for our RFCs than an actual documentation? I mean those new opportunities will be old pretty quickly :)